### PR TITLE
CPM-455: Add query to retrieve AttributeOptions

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Query/InternalApi/AttributeOption/SqlGetAttributeOptions.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Structure\Bundle\Query\InternalApi\AttributeOption;
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Structure/Bundle/Query/InternalApi/AttributeOption/SqlGetAttributeOptions.php:5]: Should be in the PublicApi

File: src/Akeneo/Pim/Structure/Component/Query/InternalApi/GetAttributeOptions.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Structure\Component\Query\InternalApi;
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Structure/Component/Query/InternalApi/GetAttributeOptions.php:4]:
```suggestion
namespace Akeneo\Pim\Structure\Component\Query\PublicApi;
```

**Review by nmarniesse — APPROVED** [review]
:+1: GTM after fix

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlGetAttributeOptions.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    private const BATCH_QUERY_SIZE = 1000;
+
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function forAttributeCode(string $attributeCode): \Iterator
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlGetAttributeOptions.php:26]:
```suggestion
    public function forAttributeCode(string $attributeCode): iterable
```

**Review by mmetayer — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlGetAttributeOptions.php (hunk 3 above)
Comment (nmarniesse) [reply src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlGetAttributeOptions.php:26]: I prefer when the return type is precise. `\Iterator` is fine to me

Comment (nmarniesse) [reply src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/AttributeOption/Sql/SqlGetAttributeOptions.php:26]: Oh sorry the interface says `iterable`. Both are ok to me

File: src/Akeneo/Pim/Structure/Component/Query/PublicApi/AttributeOption/GetAttributeOptions.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+interface GetAttributeOptions
+{
+    /**
+     * @param string $attributeCode
+     *
+     * @return AttributeOption[]
+     */
+    public function forAttributeCode(string $attributeCode): Iterable;
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Structure/Component/Query/PublicApi/AttributeOption/GetAttributeOptions.php:17]:
```suggestion
    /**
     * @return iterable<AttributeOption>
     */
    public function forAttributeCode(string $attributeCode): iterable;
```

**Review by nmarniesse — APPROVED** [review]

**Review by Balamung — COMMENTED** [review]
refresh lock

**Pull request merged**